packages/aws-cdk/test/api/deploy-stack.test.ts

```diff
@@ -423,6 +423,27 @@ test('deploy not skipped if template did not change but one tag removed', async
   expect(cfnMocks.getTemplate).toHaveBeenCalledWith({ StackName: 'withouterrors', TemplateStage: 'Original' });
 });
 
+test('deploy is not skipped if stack is in a _FAILED state', async () => {
+  // GIVEN
+  givenTemplateIs(FAKE_STACK.template);
+  givenStackExists({
+    StackStatus: 'DELETE_FAILED',
+  });
+
+  // WHEN
+  let err;
+  await deployStack({
+    stack: FAKE_STACK,
+    sdk,
+    sdkProvider,
+    resolvedEnvironment: mockResolvedEnvironment(),
+    usePreviousParameters: true,
+  }).catch(e => err = e);
+
+  // THEN
+  expect(err).toEqual(Error('The stack named withouterrors failed to deploy: DELETE_FAILED'));
```

Comment:
It's not clear to me why this is the correct test for this behavior change.

Don't we want to assert that "createChangeSet" is called, or something to that effect?

Reply:
I've pushed a change to the test asserting a call on createChangeSet, however i'm not sure that in that case the expected behaviour is fully captured by the test.

Currently, the stack will actually deploy, after which `waitForStackDeploy` will throw an error due to the stack not being in a success state.